pilot/pkg/security/authz/builder/builder.go

```diff
@@ -160,13 +174,26 @@ func (b Builder) build(policies []model.AuthorizationPolicy, action rbacpb.RBAC_
 		Action:   action,
 		Policies: map[string]*rbacpb.Policy{},
 	}
+	shadowRules := &rbacpb.RBAC{
+		Action:   action,
+		Policies: map[string]*rbacpb.Policy{},
+	}
 
 	var providers []string
 	filterType := "HTTP"
 	if forTCP {
 		filterType = "TCP"
 	}
+	hasNormalPolicy, hasDryRunPolicy := false, false
 	for _, policy := range policies {
+		var toRule *rbacpb.RBAC
```

Comment:
optional, readabilty can be slightly improved if we rename the `rules` to `normalRules` and the pointer here is kept as `rule`. It's unclear to me what this "to" refers to.

Reply:
yeah, good point, renamed to `enforceRules` to make it more clear.